### PR TITLE
Feature: upgrade keycloak

### PR DESCRIPTION
**Pull request by hill-daniel**

### CODE REVIEW

**Review by webatspeed — APPROVED** [review]

File: src/main/java/de/diakonie/onlineberatung/log/LoggingEventListenerProvider.java [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+ * LoggingEventListenerProvider logs given event with logger, but omits the IP address. See also
+ * {@link org.keycloak.events.log.JBossLoggingEventListenerProvider}.
+ */
+public class LoggingEventListenerProvider implements EventListenerProvider {
+
+  private final KeycloakSession session;
+  private final Logger logger;
+  private final Logger.Level successLevel;
+  private final Logger.Level errorLevel;
+  private final EventListenerTransaction tx = new EventListenerTransaction(this::logAdminEvent,
```
Comment (maeventer) [inline src/main/java/de/diakonie/onlineberatung/log/LoggingEventListenerProvider.java]: Could we find a better name than "tx" e.g. listenerTransaction?

File: src/main/java/de/diakonie/onlineberatung/log/LoggingEventListenerProvider.java [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+    sb.append(event.getClientId());
+    sb.append(", userId=");
+    sb.append(event.getUserId());
+
+    if (event.getError() != null) {
+      sb.append(", error=");
+      sb.append(event.getError());
+    }
+
+    if (event.getDetails() != null) {
```
Comment (maeventer) [inline src/main/java/de/diakonie/onlineberatung/log/LoggingEventListenerProvider.java:71]: Could you please extract the details, session and trace logs in several methods to shorten the lines of this method?

File: src/main/java/de/diakonie/onlineberatung/log/LoggingEventListenerProvider.java [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+      sb.append(uriInfo.getRequestUri().toString());
+    }
+
+    if (headers == null) {
+      return;
+    }
+
+    sb.append(", cookies=[");
+    boolean f = true;
+    for (Map.Entry<String, Cookie> e : headers.getCookies().entrySet()) {
```
Comment (maeventer) [inline src/main/java/de/diakonie/onlineberatung/log/LoggingEventListenerProvider.java]: Could we rename "e" to "cookie" please?

File: src/main/java/de/diakonie/onlineberatung/log/LoggingEventListenerProvider.java [hunk 4]
```diff
@@ -0,0 +1,168 @@
...
+      sb.append(", requestUri=");
+      sb.append(uriInfo.getRequestUri().toString());
+    }
+
+    if (headers == null) {
+      return;
+    }
+
+    sb.append(", cookies=[");
+    boolean f = true;
```
Comment (maeventer) [inline src/main/java/de/diakonie/onlineberatung/log/LoggingEventListenerProvider.java]: Could we also please rename "f" to something like "isFirstEntry"?

**Review by maeventer — CHANGES_REQUESTED** [review]

**Review by maeventer — APPROVED** [review]

**Pull request merged**